### ISSUE TITLE
optimizer.apply_gradients does not accept tape.gradient information

### ISSUE THREAD

**Issue by maclopes** [tensorflow/tensorflow#32478]
**System information**
Running inside colab

**Describe the current behavior**

I was expecting to be able to apply the same list of variables in apply_gradient as in t.gradient. But I get an error.

from __future__ import absolute_import, division, print_function, unicode_literals
import tensorflow as tf
tf.enable_eager_execution()
optimizer = tf.train.GradientDescentOptimizer(0.001)
x = tf.Variable(2*tf.ones((5,1)))  
with tf.GradientTape(persistent=True) as t:
  y = x**2
  dy_dx = t.gradient(y, x)
  optimizer.apply_gradients(dy_dx,x)  # optimizer = Adam

**Comment by oanush** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
